java/ql/lib/semmle/code/java/security/SensitiveLoggingQuery.qll

```diff
@@ -4,14 +4,15 @@ import java
 private import semmle.code.java.dataflow.ExternalFlow
 import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.security.SensitiveActions
+import semmle.code.java.frameworks.android.Compose
 import DataFlow
 
-/** A variable that may hold sensitive information, judging by its name. * */
+/** A variable that may hold sensitive information, judging by its name. */
 class CredentialExpr extends Expr {
   CredentialExpr() {
     exists(Variable v | this = v.getAnAccess() |
-      v.getName().regexpMatch([getCommonSensitiveInfoRegex(), "(?i).*(username).*"]) and
-      not v.isFinal()
+      v.getName().regexpMatch(getCommonSensitiveInfoRegex()) and
+      not (v.isFinal() and v.isStatic())
```

Comment:
Would `... and not this instanceof CompileTimeConstantExpr` get what you want? That already includes clauses

```
      // Access to a final variable initialized by a compile-time constant.
      exists(Variable v | this = v.getAnAccess() |
        v.isFinal() and
        v.getInitializer().isCompileTimeConstant()
      )
      or
      this instanceof LiveLiteral
```

Reply:
Oh, good catch! I think we still need the `LiveLiteral` sanitizer though, because there are other elements with awful names being generated by the Compose compiler plugin that don't fit in the `LiveLiteral` class but flow through it.

But this is still cleaner for the "not a constant" check, so I'll add it, thanks.